packages/libs/react-ui/src/components/Card/Card.css.ts

```diff
@@ -0,0 +1,82 @@
+import { sprinkles, vars } from '../../styles';
+
+import { style, styleVariants } from '@vanilla-extract/css';
+
+export const container = style([
+  sprinkles({
+    background: 'neutral2',
+    color: 'neutral6',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'flex-start',
+    padding: 'lg',
+    borderRadius: 'sm',
+  }),
+]);
+
+const booleans: Record<string, boolean> = {
+  true: true,
+  false: false,
+};
+
+export const fullWidthVariant = styleVariants(booleans, (fullWidth) => {
+  return [
+    container,
+    sprinkles({
+      width: fullWidth ? 'full' : 'max-content',
+    }),
+  ];
+});
+
+export const stackVariant = styleVariants(booleans, (stack) => {
+  const selectors = {
+    '&:not(:last-child)': {
+      borderBottom: `1px solid ${vars.colors.neutral3}`,
+    },
+    '&:first-child': {
+      borderRadius: `${vars.radii.sm} ${vars.radii.sm} 0 0`,
+    },
+    '&:last-child': {
+      borderRadius: `0 0 ${vars.radii.sm} ${vars.radii.sm}`,
+    },
+  };
+
+  return [
+    container,
+    sprinkles(
+      stack
+        ? {
+            marginY: 0,
+          }
+        : {
+            marginY: 'md',
+            borderRadius: 'sm',
+            border: 'none',
+          },
+    ),
+    {
+      selectors: stack ? selectors : {},
+    },
+  ];
+});
```

Comment:
I'm thinking that with vanilla extract the use of variants for booleans seems messier. I would recommend using classnames and styling the default (false styles) as part of the main "style" and with the additional ones optionally added to override the defaults. You can use this library as a utility to do this  https://www.npmjs.com/package/classnames

```
<div className={classNames(base, { stack, fullWidth })}>
```

This way the base styles + the false versions of the stack and fullWidth variants will be applied by default and if the props are true, the other styles will override the default